pkg/util/watchdog/watchdog.go

```diff
@@ -22,12 +22,10 @@ func (w *SimpleWatchdog) Start(stop <-chan struct{}) {
 	ticker := w.NewTicker()
 	defer ticker.Stop()
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	for {
-		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
this breaks the fix. we need a new context every tick. can we just wait on two channels in the goroutine? and at the end of the `for` loop, we close the new channel? that way the goroutine returns

Reply:
If that breaks probably waiting for 2 channels is the best options